go_modules/spec/dependabot/go_modules/file_updater_spec.rb

```diff
@@ -148,5 +162,91 @@
         expect(updated_files.find { |f| f.name == "go.sum" }).to_not be_nil
       end
     end
+
+    context "vendoring" do
+      let(:project_name) { "vendor" }
+      let(:vendor) { true }
+
+      let(:dependency_name) { "github.com/pkg/errors" }
+      let(:dependency_version) { "v0.9.1" }
+      let(:dependency_previous_version) { "v0.8.0" }
+      let(:requirements) do
+        [{
+          file: "go.mod",
+          requirement: dependency_version,
+          groups: [],
+          source: {
+            type: "default",
+            source: "github.com/pkg"
+          }
+        }]
+      end
+      let(:previous_requirements) do
+        [{
+          file: "go.mod",
+          requirement: dependency_previous_version,
+          groups: [],
+          source: {
+            type: "default",
+            source: "github.com/pkg"
+          }
+        }]
+      end
+
+      it "updates the version" do
+        expect(go_mod_body).to include("github.com/pkg/errors v0.8.0")
+
+        updater.updated_dependency_files
+
+        go_mod_file = updater.updated_dependency_files.find do |file|
+          file.name == "go.mod"
+        end
+
+        expect(go_mod_file.content).to include "github.com/pkg/errors v0.9.1"
+      end
+
+      it "includes the vendored files" do
+        expect(updater.updated_dependency_files.map(&:name)).to match_array(
+          %w(
+            go.mod
+            go.sum
+            vendor/github.com/pkg/errors/.travis.yml
+            vendor/github.com/pkg/errors/Makefile
+            vendor/github.com/pkg/errors/README.md
+            vendor/github.com/pkg/errors/errors.go
+            vendor/github.com/pkg/errors/go113.go
+            vendor/github.com/pkg/errors/stack.go
+            vendor/modules.txt
+          )
+        )
+      end
+
+      it "updates the vendor/modules.txt file to the right version" do
+        modules_file = updater.updated_dependency_files.find do |file|
+          file.name == "vendor/modules.txt"
+        end
+
+        expect(modules_file.content).to include "github.com/pkg/errors v0.9.1"
```

Comment:
```suggestion
        expect(modules_file.content).to_not include "github.com/pkg/errors v0.8.0"
        expect(modules_file.content).to include "github.com/pkg/errors v0.9.1"
```

I think? (and I think this is independent of `vendoring`  - there can be only one)